src/metadata.cc

```diff
@@ -267,50 +285,85 @@ Napi::Object Metadata::Init(Napi::Env env, Napi::Object exports) {
 }
 
 //
-// function to format an x-trace with components split by ':'
+// function to format an x-trace with components split by sep.
 //
-bool Metadata::format(oboe_metadata_t* md, size_t len, char* buffer) {
+bool Metadata::format(oboe_metadata_t* md, size_t len, char* buffer, uint flags, char sep) {
     char* b = buffer;
+    char base = (flags & Metadata::ff_lowercase ? 'a' : 'A') - 10;
+
+    auto puthex = [&b, base](uint8_t byte) {
+      int digit = (byte >> 4);
+      digit += digit <= 9 ? '0' : base;
+      *b++ = (char) digit;
+      digit = byte & 0xF;
+      digit += digit <= 9 ? '0' : base;
+      *b++ = (char) digit;
+
+      return b;
+    };
```

Comment:
i do use oboe_metadata_tostr() when it's just plan vanilla. [here](https://github.com/appoptics/appoptics-bindings-node/blob/f0900ffa75ba93c0f51723baf02d835248755921/src/metadata.cc#L188). i created format initially because i wanted to split the fields (2b, task, op, flags) with a character to make looking at them easier. it was especially helpful when i was first learning and doing a lot of debugging. but i didn't want to go through the sequence of function calls (3 splits) followed by concatenating the four pieces back together just to insert delimiters. when it turns out we needed another format for logs i decided to generalize the format function so it could handle almost any combination of needs. it's also what let me play around with and learn the basics of lambda functions in c++, so it was kind of fun too. format can now actually produce exactly what oboe does but i still call oboe in the no argument case because i want to be able to verify that both are doing the same thing.

i actually spent too long trying to get enum {} to work for the flags but it required overloading operators and a bunch of work that practically had little benefit. i thought that would be pretty easy but it turns out it's not.